### DISCUSSION TITLE
Facing issues while using graphql-modules with Apollo Server V4

### DISCUSSION THREAD

**Discussion by sourabh-kumar** [Urigo/graphql-modules#2286] [Q&A] [answered]
I'm creating a GraphQL application using Apollo Server V4 and graphql-modules.

This is how I'm initialising the app
```js
import { createApplication } from "graphql-modules";

import moduleA from "./moduleA";
import moduleB from "./moduleB";

export const gqlApp = createApplication({
    modules: [moduleA, moduleB]
});
```

And I'm initialising the Apollo server in the following way:
```js
 const server = new ApolloServer<ContextValue>({
        schema: gqlApp.schema,
        plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
        includeStacktraceInErrorResponses: "production" !== process.env.NODE_ENV
    });
```

When I run the app, I get the following error in the response
`context.ɵgetModuleContext is not a function`

The only way I can make this work right now is if I generate my schema the following way:
`schema: gqlApp.createSchemaForApollo()`

But this seems to be marked as deprecated.

I'd appreciate if someone can help me in figuring out the right way of doing this.

**Comment by ardatan** [comment, marked as answer]
See here https://github.com/Urigo/graphql-modules/issues/2270

**Reply by sourabh-kumar** [reply]
This works. Thanks @ardatan 

Even tested with the [mocking mechanism recommended by Apollo](https://www.apollographql.com/docs/apollo-server/testing/mocking/)

```js
//.....

   const gqlSchemaWithMocks = addMocksToSchema({
        schema: gqlApp.schema,
        mocks: {
            ...someMock
        },
        preserveResolvers: true
    });

const server = new ApolloServer<ContextValue>({
        gateway: {
            async load() {
                return { executor: gqlApp.createApolloExecutor() };
            },
            onSchemaLoadOrUpdate(callback) {
                callback({ apiSchema: gqlSchemaWithMocks } as any);
                return () => {};
            },
            async stop() {}
        },
        plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
        includeStacktraceInErrorResponses: "production" !== process.env.NODE_ENV
    });

//.....
````

There  is mention of the [gateway and executor in Apollo's documentation](https://www.apollographql.com/docs/apollo-server/migration/#executor) and the [graphql-module's official documentation explains it w.r.t Apollo V3's executor implementation](https://the-guild.dev/graphql/modules/docs/get-started#use-your-application).

But neither of the docs talk about this way of adding the gateway.

It'd be good to get this added to the official docs.
